libraries/AP_IOMCU/AP_IOMCU.cpp

```diff
@@ -1198,15 +1198,19 @@ bool AP_IOMCU::setup_mixing(RCMapper *rcmap, int8_t override_chan,
         MIX_UPDATE(mixing.servo_reversed[i], c->get_reversed());
     }
     // update RCMap
-    MIX_UPDATE(mixing.rc_channel[0], rcmap->roll());
-    MIX_UPDATE(mixing.rc_channel[1], rcmap->pitch());
-    MIX_UPDATE(mixing.rc_channel[2], rcmap->throttle());
-    MIX_UPDATE(mixing.rc_channel[3], rcmap->yaw());
-    for (uint8_t i=0; i<4; i++) {
-        const RC_Channel *c = RC_Channels::rc_channel(mixing.rc_channel[i]-1);
-        if (!c) {
+    static const RC_Channel::AUX_FUNC funcs[] {
+        RC_Channel::AUX_FUNC::ROLL,
+        RC_Channel::AUX_FUNC::PITCH,
+        RC_Channel::AUX_FUNC::YAW,
+        RC_Channel::AUX_FUNC::THROTTLE,
+    };
+    for (uint8_t i=0; i<ARRAY_SIZE(funcs); i++) {
+        const RC_Channel *c = rc().find_channel_for_option(funcs[i]);
```

Comment:
I guess you've inspected each place we use the new find_channel_for_option() to be sure it checks for nullptr?

Reply:
This is actually an existing problem with the codebase; if you set a bad value in rcmap you're not going to go anywhere until you wipe your parameters by uploading a different vehicle.  I've just checked in SITL, and *it* certainly  behaves that way.  This is because `radio.cpp` just unconditionaly sets the pointers from the rcmap values and *nobody* wants to check those RC channel pointers for nullptr every time they're looked at!

I don't think this PR makes the problem worse, except that the RCn_OPTIONS are much more in-your-face as a user.

I think we should address the problem, but it's a little tricky as we like to be able to reset RCn_OPTION parameters and have them *generally* start to work.  And certainly want that behaviour for RC input channels IMO!

So what I propose is that we only update the RC input channel pointers when disarmed, and if they *do* go to nullptr then go to the config error loop.  And for always-armed planes/rovers/trackers we do the assignments exactly once while armed.